### PR TITLE
Adding on flatTransform to OptionT as suggested. (Issue #2309)

### PR DESCRIPTION
**Pull request by abhishek7**
#2309

### CODE REVIEW

File: tests/src/test/scala/cats/tests/OptionTSuite.scala [hunk 1]
```diff
@@ -290,6 +290,12 @@ class OptionTSuite extends CatsSuite {
     }
   }
 
+  test("flatTransform consistent with value.map") {
```
Comment (kubukoz) [inline tests/src/test/scala/cats/tests/OptionTSuite.scala]: should be `with value.flatMap`

**Comment by kubukoz** [comment]
Do you think we should add this for `EitherT` and `IorT` in some way too? Once this is merged we could prepare additional PRs for them.

I'm still not sure about the name, so I'd wait for one of the maintainers to take a look and decide.

File: tests/src/test/scala/cats/tests/OptionTSuite.scala (hunk 1 above)
Comment (abhishek7) [reply tests/src/test/scala/cats/tests/OptionTSuite.scala]: Good catch, will update.

**Comment by abhishek7** [comment]
@kubukoz I think we should, and I do agree with you that it would make sense to have one of the maintainers decide. I also agree that separate PRs for each would be the way to go if the decision is to go ahead with it.

**Comment by abhishek7** [comment]
@kubukoz, any particular maintainers we should request to take a look at this?

**Comment by LukaJCB** [comment]
I think `flatTransform` is probably fine in terms of naming :)

**Comment by abhishek7** [comment]
@kubukoz @LukaJCB friendly ping on this.

**Comment by kubukoz** [comment]
I'm not a maintainer so I'll just delegate to Luka ;)

**Comment by abhishek7** [comment]
Thanks @LukaJCB! It appears that I don't have the ability to merge. @kailuowang, @LukaJCB, not sure if you're able to?

**Pull request merged**